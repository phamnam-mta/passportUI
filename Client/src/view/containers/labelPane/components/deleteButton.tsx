import * as React from "react";
import { IconButton } from "@fluentui/react";

export const DeleteButton = (props: { onClick: (event) => void; disabled?: boolean }): JSX.Element => {
    return (
        <IconButton
            title="Delete"
            iconProps={{ iconName: "Cancel" }}
            styles={{
                icon: { fontSize: 14 },
                root: { width: 24, height: 24 },
                rootDisabled: { backgroundColor: "transparent" },
            }}
            onClick={props.onClick}
            disabled={props.disabled}
        />
    );
};
