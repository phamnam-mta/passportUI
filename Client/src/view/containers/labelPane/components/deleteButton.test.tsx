import * as React from "react";
import { shallow } from "enzyme";
import { DeleteButton } from "./deleteButton";

describe("<DeleteButton />", () => {
    let baseProps;
    beforeEach(() => {
        baseProps = {
            onClick: jest.fn(),
            disabled: false,
        };
    });

    describe("Rendering", () => {
        it("should match snapshot, when delete button is enabled", () => {
            const wrapper = shallow(<DeleteButton {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when delete button is disabled", () => {
            const props = { ...baseProps, disabled: true };
            const wrapper = shallow(<DeleteButton {...props} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should trigger onClick, when delete button is clicked", () => {
            const wrapper = shallow(<DeleteButton {...baseProps} />);
            wrapper.simulate("click");

            expect(baseProps.onClick).toBeCalledTimes(1);
        });
    });
});
