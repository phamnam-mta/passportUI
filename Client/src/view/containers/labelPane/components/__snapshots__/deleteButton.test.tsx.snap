// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> Rendering should match snapshot, when delete button is disabled 1`] = `
<CustomizedIconButton
  disabled={true}
  iconProps={
    Object {
      "iconName": "Cancel",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "icon": Object {
        "fontSize": 14,
      },
      "root": Object {
        "height": 24,
        "width": 24,
      },
      "rootDisabled": Object {
        "backgroundColor": "transparent",
      },
    }
  }
  title="Delete"
/>
`;

exports[`<DeleteButton /> Rendering should match snapshot, when delete button is enabled 1`] = `
<CustomizedIconButton
  disabled={false}
  iconProps={
    Object {
      "iconName": "Cancel",
    }
  }
  onClick={[MockFunction]}
  styles={
    Object {
      "icon": Object {
        "fontSize": 14,
      },
      "root": Object {
        "height": 24,
        "width": 24,
      },
      "rootDisabled": Object {
        "backgroundColor": "transparent",
      },
    }
  }
  title="Delete"
/>
`;
